### DISCUSSION TITLE
File Download

### DISCUSSION THREAD

**Discussion by yeaharh** [microsoft/typespec#4320] [Q&A] [answered]
Hi

We're building an OpenAPI compliant API that will return a pdf file from a get request.
Is there an example we can look at?

Rgds
Jim

**Comment by timotheeguerin** [comment, marked as answer]
You can do that by setting the content type and the body to be `bytes`
```ts
import "@typespec/http";

using TypeSpec.Http;

op download(): {
  @header contentType: "application/pdf";
  @body file: bytes;
};
```
[Playground](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwoKb3AgZG93bmxvYWQoKTogewogIEBoZWFkZXIgY29udGVudMQyOiAiYXBwbGljYXRpb24vcGRmIjvEKmJvZHkgZmlsZTogYnl0ZXM7Cn07Cg%3D%3D&e=%40typespec%2Fopenapi3&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40typespec%2Fhttp%2Fall%22%5D%7D%7D)

**Reply by yeaharh** [reply]
Thanks you.

Is it possible to return (download) both the pdf and a json payload.  Something like a multipart/form-data upload.

**Reply by timotheeguerin** [reply]
not sure what you mean, can you share what openapi you would like to see?

**Reply by yeaharh** [reply]
We have a file upload API which is using the multipart/form-data content type.  This allows a file to be uploaded as one part and json in another part.

@doc("ScanRequest Payload")
model ScanRequest {

  // Request Headers
  // api version header
  @doc("api version header")
  @header apiVersion: apiVersionNumber;
  // content-type header
  @header contentType: "multipart/form-data";
  // tenantName header associated with the scan request
  @doc("tenantName header")
  @header tenantName: string;

  // Request Query Parameters
  @doc("Unique identifier for the scan request")
  @format("uuid")
  @query scanRequestId?: string;

  // Request Body
  @doc("filename of file to scan")
  fileName: string;
  @doc("file content to scan")
  @format("binary")
  fileToScan: string;
}

Just wondering if the same (multipart) approach is possible with a download.
